Objects/abstract.c

```diff
@@ -495,6 +494,47 @@ _Py_add_one_to_index_C(int nd, Py_ssize_t *index, const Py_ssize_t *shape)
     }
 }
 
+Py_ssize_t
+PyBuffer_SizeFromFormat(const char *format)
+{
+    PyObject *structmodule = NULL;
+    PyObject *Struct = NULL;
+    PyObject *res = NULL;
+    PyObject *fmt = NULL;
+    Py_ssize_t itemsize = -1;
+
+    structmodule = PyImport_ImportModule("struct");
```

Comment:
You are leaking one reference to structmodule: Py_DECREF(structmodule) somewhere is needed.

You can check for refleak using: ./python -m test -R 3:3 test_buffer -m test_pybuffer_size_from_format